Automattic/wp-calypso
# WPCOM Block Editor: Add site editor save tracks event

#### Changes proposed in this Pull Request

* Add a site editor specific tracks event when the editor save button is clicked

#### Testing instructions



1. Sandbox a site and wp widgets
2. Navigate to the site editor
3. Make modifications to a template part
4. Open the dev console and switch to the `Network Requests` tab
5. Filter requests by `t.gif` 
6. Save the changes to the editor
7. Verify that a tracking `t.gif` event was sent for `wpcom_block_editor_site_editor_save_click`

#### Screenshots
<img width="1654" alt="Screen Shot 2022-01-24 at 11 30 25 AM" src="https://user-images.githubusercontent.com/5414230/150851200-fc337bc4-0362-4f1c-878c-40db9c01a3e9.png">



Fixes https://github.com/Automattic/wp-calypso/issues/60254
